src/app/tests/TestThreadBorderRouterManagementCluster.cpp

```diff
@@ -127,39 +116,100 @@ class TestDelegate : public Delegate
+                                                                      0x88, 0x99, 0xAA, 0xBB, 0xCC, 0xDD, 0xEE, 0xFF };
+    const uint16_t kTestThreadVersion                             = 4;
+    uint8_t mActiveDataset[Thread::kSizeOperationalDataset]       = { 0 };
+    size_t mActiveDatasetLen                                      = 0;
+    uint8_t mStoredActiveDataset[Thread::kSizeOperationalDataset] = { 0 };
+    size_t mStoredActiveDatasetLen                                = 0;
+    uint8_t mPendingDataset[Thread::kSizeOperationalDataset]      = { 0 };
+    size_t mPendingDatasetLen                                     = 0;
+    bool mUseInvalidBorderRouterName                              = true;
+    size_t mTestBorderAgentIdLen                                  = kBorderAgentIdLength - 1;
+    bool mInterfaceEnabled                                        = false;
+    uint32_t mSetActiveDatasetCommandSequenceNum                  = 0;
+    ActivateDatasetCallback * mCallback                           = nullptr;
 };
 
-constexpr chip::EndpointId sTestEndpointId    = 1;
-static TestDelegate * sTestDelegatePtr        = nullptr;
-static ServerInstance * sTestSeverInstancePtr = nullptr;
+constexpr EndpointId kTestEndpointId            = 1;
+constexpr FabricIndex kTestAccessingFabricIndex = 1;
+static FailSafeContext sTestFailsafeContext;
+static TestDelegate sTestDelegate;
+static ServerInstance sTestSeverInstance(kTestEndpointId, &sTestDelegate, sTestFailsafeContext);
+
+class TestSetActiveDatasetCommandHandler : public CommandHandler
+{
+public:
+    TestSetActiveDatasetCommandHandler() : mClusterStatus(Protocols::InteractionModel::Status::Success) {}
+    CHIP_ERROR FallibleAddStatus(const ConcreteCommandPath & aRequestCommandPath,
+                                 const Protocols::InteractionModel::ClusterStatusCode & aStatus, const char * context = nullptr)
+    {
+        return CHIP_NO_ERROR;
+    }
+
+    void AddStatus(const ConcreteCommandPath & aRequestCommandPath, const Protocols::InteractionModel::ClusterStatusCode & aStatus,
+                   const char * context = nullptr)
+    {
+        mClusterStatus = aStatus;
+    }
+
+    FabricIndex GetAccessingFabricIndex() const { return kTestAccessingFabricIndex; }
+
+    CHIP_ERROR AddResponseData(const ConcreteCommandPath & aRequestCommandPath, CommandId aResponseCommandId,
+                               const DataModel::EncodableToTLV & aEncodable)
+    {
+        return CHIP_NO_ERROR;
+    }
+
+    void AddResponse(const ConcreteCommandPath & aRequestCommandPath, CommandId aResponseCommandId,
+                     const DataModel::EncodableToTLV & aEncodable)
+    {}
+
+    bool IsTimedInvoke() const { return false; }
+
+    void FlushAcksRightAwayOnSlowCommand() {}
+
+    Access::SubjectDescriptor GetSubjectDescriptor() const
+    {
+        Access::SubjectDescriptor subjectDescriptor;
```

Comment:
This object is uninitialized. It should be initialized

Reply:
Done